WordPress/openverse-frontend
# Deploy production frontend on release & correctly time staging deployment

## Fixes
Fixes #1557 by @sarayourfriend 

## Description


Deploys https://search-production.openverse.engineering whenever a release is cut in this repository. At least hopefully! We'll need to see if the GHCR configuration actually works for releases and if it will successfully trigger the "deploy-production" workflow to completion with the release tag. Unfortunately I'm not aware of any way to test this functionality other than by merging it and see if it works with a test release.

I also noticed that the staging deployment and the GHCR push to the `main` tag have a race condition. I've fixed that in this PR by only running the staging deployment after the GHCR image is uploaded after a push to the `main` branch.

Also: I've removed the old push_prod and push_staging workflows that were used to push images to ECR. We no longer use ECR, so those workflows will fail until they're removed.

Also: this adds the `version.json` requested by the Nginx sidecar issue. I think we can safely close that issue for now and in the future look into whether we want to use Nginx to serve static files. We don't currently do that, and the version.json endpoint isn't worth the extra complexity for now.

## Testing Instructions

I think just merge and see how it goes. Note that the existing GHCR push should still work for this branch. You can see I did a bunch of test commits to get the action changes verified as working, in particular the tag extraction for the RELEASE build arg.

To test the build arg, checkout this branch and run:

```
docker build . --build-arg RELEASE=super-sweet-release-sha-1234abcd -t openverse-frontend:latest
```

Run the generated image and verify that you can reach the `/version.json` endpoint at `http://localhost:8443/version.json`:

```
docker run -p 8443:8443/tcp openverse-frontend:latest
```

## Checklist

- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
